pkg/scheduling/v2/slot.go

```diff
@@ -186,11 +186,13 @@ func getRankedSlots(
 			continue
 		}
 
-		// if this is a HARD sticky strategy, it can only be assigned to the desired worker if the desired
-		// worker id is set. otherwise, it cannot be assigned.
+		// if this is a HARD sticky strategy, and there's a desired worker id, it can only be assigned to that
+		// worker. if there's no desired worker id, we assign to any worker.
 		if qi.Sticky.Valid && qi.Sticky.StickyStrategy == dbsqlc.StickyStrategyHARD {
 			if qi.DesiredWorkerId.Valid && workerId == sqlchelpers.UUIDToStr(qi.DesiredWorkerId) {
 				validSlots.addSlot(slot, 0)
+			} else if !qi.DesiredWorkerId.Valid {
```

Comment:
nit (not blocking): can we use an or in the original condition? e.g.:

```suggestion
			if (qi.DesiredWorkerId.Valid && workerId == sqlchelpers.UUIDToStr(qi.DesiredWorkerId)) || !qi.DesiredWorkerId.Valid {
				validSlots.addSlot(slot, 0)
			}
```

Reply:
Yes, that's better, though I think it makes it more difficult to reason about -- probably something like this would work best?

```go
matchesDesired := qi.DesiredWorkerId.Valid && workerId == sqlchelpers.UUIDToStr(qi.DesiredWorkerId)
desiredNotSet := !qi.DesiredWorkerId.Valid

if matchesDesired || desiredNotSet {
  validSlots.addSlot(slot, 0)
}
```

Reply:
yeah, that looks good to me 👍